lib/iris/tests/test_pandas.py

```diff
@@ -491,13 +502,433 @@ def test_copy_true(self):
         data_frame = pandas.DataFrame([[0, 1, 2, 3, 4], [5, 6, 7, 8, 9]])
         cube = iris.pandas.as_cube(data_frame)
         cube.data[0, 0] = 99
-        self.assertEqual(data_frame[0][0], 0)
+        assert data_frame[0][0] == 0
 
     def test_copy_false(self):
         data_frame = pandas.DataFrame([[0, 1, 2, 3, 4], [5, 6, 7, 8, 9]])
         cube = iris.pandas.as_cube(data_frame, copy=False)
         cube.data[0, 0] = 99
-        self.assertEqual(data_frame[0][0], 99)
+        assert data_frame[0][0] == 99
+
+
+@skip_pandas
+class TestFutureAndDeprecation(tests.IrisTest):
+    def test_future_error(self):
+        iris.FUTURE.pandas_ndim = True
+        data_frame = pandas.DataFrame([[0, 1, 2, 3, 4], [5, 6, 7, 8, 9]])
+        with pytest.raises(
+            RuntimeError, match="does not support n-dimensional"
+        ):
+            _ = iris.pandas.as_cube(data_frame)
+
+    def test_deprecation_warning(self):
+        data_frame = pandas.DataFrame([[0, 1, 2, 3, 4], [5, 6, 7, 8, 9]])
+        with pytest.warns(
+            IrisDeprecation, match="as_cube has been deprecated"
+        ):
+            _ = iris.pandas.as_cube(data_frame)
+
+
+@skip_pandas
+class TestPandasAsCubes(tests.IrisTest):
+    @staticmethod
+    def _create_pandas(index_levels=0, is_series=False):
+        index_length = 3
+
+        index_names = [f"index_{i}" for i in range(index_levels)]
+        index_values = [
+            np.arange(index_length) * 10 * (i + 1) for i in range(index_levels)
+        ]
+
+        if index_levels == 1:
+            index = pandas.Index(index_values[0], name=index_names[0])
+            data_length = index_length
+        elif index_levels > 1:
+            index = pandas.MultiIndex.from_product(
+                index_values, names=index_names
+            )
+            data_length = index.nunique()
+        else:
+            index = None
+            data_length = index_length
+
+        data = np.arange(data_length) * 10
+
+        if is_series:
+            class_ = pandas.Series
+        else:
+            class_ = pandas.DataFrame
+
+        return class_(data, index=index)
+
+    def test_1d_no_index(self):
+        df = self._create_pandas()
+        result = iris.pandas.as_cubes(df)
+
+        expected_coord = DimCoord(df.index.values)
+        expected_cube = Cube(
+            data=df[0].values,
+            long_name=str(df[0].name),
+            dim_coords_and_dims=[(expected_coord, 0)],
+        )
+        assert result == [expected_cube]
+
+    def test_1d_with_index(self):
+        df = self._create_pandas(index_levels=1)
+        result = iris.pandas.as_cubes(df)
+
+        expected_coord = DimCoord(df.index.values, long_name=df.index.name)
+        result_cube = result[0]
```

Comment:
You could make this (and the other places you've used this pattern in the tests)
```
result_cube, = result
```
So that if there are more than one cubes it raises an error and thus fails the test?

Reply:
Fantastic idea, thanks!

dfabf35762b4b43fd3eab0cebcdf131619492bbc